### PR TITLE
Add opt-in forecast risk scoring diagnostic

### PR DESCRIPTION
**Pull request by ll7**
## Summary by CodeRabbit

* **New Features**
  * Added forecast risk weighting capability to policy stack, enabling configurable risk weight and observation parameters for decision-making
  * Added diagnostic validation tools for evaluating and comparing forecast risk behavior

* **Tests**
  * Added comprehensive test coverage for forecast risk functionality including configuration parsing, behavioral impact, and edge cases
  * Added evidence documentation for diagnostic comparison validation

### CODE REVIEW

File: scripts/validation/validate_forecast_risk_policy_stack.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    observation: dict[str, Any],
+    *,
+    risk_dwa: _FixedRiskDWA,
+) -> dict[str, Any]:
+    """Run one adapter step and return the last-step diagnostic packet."""
+    stack = PolicyStackV1Adapter(config=config, risk_dwa=risk_dwa)
+    stack.plan(observation)
+    last = stack.diagnostics()["last_step"]
+    assert last is not None, "diagnostics last_step must exist after plan()"
+    return last
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/validation/validate_forecast_risk_policy_stack.py:102]: Addressed in b0bbe84cb632e667eec7def311c593f07d034543: the validation helper now uses an explicit if last is None check that raises AssertionError instead of relying on assert, and the targeted validation plus final PR readiness passed.

File: robot_sf/planner/policy_stack_v1.py [hunk 2]
```diff
@@ -438,14 +453,37 @@ def _score_command(
...
             "heading_alignment": float(heading_alignment),
             "ped_clearance": float(clearance_bonus),
             "angular_penalty": float(angular_penalty),
         }
 
+        fr_weight = float(self.config.forecast_risk_weight)
+        if fr_weight > 0.0:
+            fr_key = self.config.forecast_risk_observation_key
+            payload = observation.get(fr_key) if isinstance(observation.get(fr_key), dict) else None
+            raw_risk, effective_risk, false_positive, available = _forecast_risk_values(payload)
```
[the remark this answers is not in the document]
Comment (ll7) [reply robot_sf/planner/policy_stack_v1.py:473]: Addressed in b0bbe84cb632e667eec7def311c593f07d034543: arbitration_trace_packet() now appends the configured forecast_risk_observation_key to inference_available_features when forecast_risk_weight > 0, exposes forecast_risk_observation_key and forecast_risk_active_statuses in observation_contract, and tests the enabled/custom-key contract.
